File: problems/0003_longest-substring-without-repeating-characters/index-of/Solution.kt
class Solution {
    fun lengthOfLongestSubstring(s: String): Int {
        var max = 0
        var start = 0
        var end = 0
        s.forEach { c ->
            val idx = indexOf(s, c, start, end)
            ++end
            if (idx == null)
                max = Math.max(max, end - start)
            else
                start = idx + 1
        }
        return max
    }
    inline fun indexOf(s: String, c: Char, start: Int, end: Int): Int? {
        for (i in start..(end - 1))
            if (s[i] == c)
                return i
        return null
    }
}

File: problems/0003_longest-substring-without-repeating-characters/index-of/Solution.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int max = 0, start = 0, end = 0;
        foreach (char c in s) {
            int count = (end++) - start,
                idx = s.IndexOf(c, start, count);
            if (idx == -1)
                max = Math.Max(max, count + 1);
            else
                start = idx + 1;
        }
        return max;
    }
}
